management/guardian.go

```diff
@@ -0,0 +1,114 @@
+package management
+
+const (
+	DuoGuardianFactor              GuardianFactorType = "duo"
+	EmailGuardianFactor            GuardianFactorType = "email"
+	OtpGuardianFactor              GuardianFactorType = "otp"
+	PushNotificationGuardianFactor GuardianFactorType = "push-notification"
+	SmsGuardianFactor              GuardianFactorType = "sms"
+)
+
+type GuardianFactorType string
+
+type GuardianFactor struct {
+	// States if this factor is enabled
+	Enabled *bool `json:"enabled,omitempty"`
+}
+
+type GuardianFactorStatus struct {
+	// States if this factor is enabled
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Guardian Factor name
+	Name *string `json:"name,omitempty"`
+
+	// For factors with trial limits (e.g. SMS) states if those limits have been exceeded
+	TrialExpired *bool `json:"trial_expired,omitempty"`
+}
```

Comment:
I think we should combine `GuardianFactor` and `GuardianFactorStatus` so that the update and list operation uses the same data structure, consistent with the rest of the codebase.